firefox-ios/Client/Coordinators/ShareSheetCoordinator.swift

```diff
@@ -77,20 +81,45 @@ class ShareSheetCoordinator: BaseCoordinator,
 
     private func handleShareSheetCompletion(
         activityType: UIActivity.ActivityType?,
-        url: URL
+        shareType: ShareType,
+        relatedTab: Tab?
     ) {
+        // NOTE: didFinish will be called on each of these code paths to properly dismiss the share sheet.
```

Comment:
This function is still messy, but because of FXIOS-10334 I was advised not to change anything.